components/post_view/post_info/post_info.tsx

```diff
@@ -21,116 +23,125 @@ import PostReaction from 'components/post_view/post_reaction';
-         */
-        isMobile: PropTypes.bool,
-
-        /**
-         * Set to render in compact view
-         */
-        compactDisplay: PropTypes.bool,
-
-        /**
-         * Set to mark post as being hovered over
-         */
-        hover: PropTypes.bool.isRequired,
-
-        /**
-         * Set to render the post time when not hovering
-         */
-        showTimeWithoutHover: PropTypes.bool.isRequired,
-
-        /**
-         * Whether to show the emoji picker.
-         */
-        enableEmojiPicker: PropTypes.bool, // Made it optional until migrating this file to TS
+export type Props = {
+
+    /**
+     * The post to render the info for
+     */
+    post: Post,
+
+    /**
+     * The id of the team which belongs the post
+     */
+    teamId?: string;
+
+    /**
+     * Function called when the comment icon is clicked
+     */
+    handleCommentClick: React.EventHandler<React.MouseEvent>;
+
+    /**
+     * Function called when the card icon is clicked
+     */
+    handleCardClick: (post: Post) => void;
+
+    /**
+     * Function called when the post options dropdown is opened
+     */
+    handleDropdownOpened: (e: boolean) => void;
+
+    /**
+     * Set to mark the post as flagged
+     */
+    isFlagged?: boolean;
+
+    /**
+     * Set to mark the post as open the extra info in the rhs
+     */
+    isCardOpen?: boolean;
+
+    /**
+     * The number of replies in the same thread as this post
+     */
+    replyCount?: number;
+
+    /**
+     * Set to indicate that this is previous post was not a reply to the same thread
+     */
+    isFirstReply?: boolean;
+
+    /**
+     * Set to render in mobile view
+     */
+    isMobile?: boolean;
+
+    /**
+     * Set to render in compact view
+     */
+    compactDisplay?: boolean;
+
+    /**
+     * Set to mark post as being hovered over
+     */
+    hover: boolean;
+
+    /**
+     * Set to render the post time when not hovering
+     */
+    showTimeWithoutHover: boolean;
+
+    /**
+     * Whether to show the emoji picker.
+     */
+    enableEmojiPicker?: boolean;
+
+    /**
+     * Set not to allow edits on post
+     */
+    isReadOnly: boolean | null;
+
+    /**
+     * To check if the state of emoji for last message and from where it was emitted
+     */
+    shortcutReactToLastPostEmittedFrom?: string;
+
+    /**
+     * To Check if the current post is last in the list
+     */
+    isLastPost?: boolean;
+
+    actions: {
 
         /**
-         * Set not to allow edits on post
+         * Function to remove the post
          */
-        isReadOnly: PropTypes.bool,
+        removePost: (post: ExtendedPost) => void;
 
         /**
-         * To check if the state of emoji for last message and from where it was emitted
+         * Function to set or unset emoji picker for last message
          */
-        shortcutReactToLastPostEmittedFrom: PropTypes.string,
+        emitShortcutReactToLastPostFrom?: (emittedFrom: string) => void;
+    } | null;
```

Comment:
Definitely shouldn't need the `|` null here. When using this component, it should always be imported by its module thus including the redux portion. The only exception is the case of the tests, where the `actions` portion should be mocked out.

Reply:
Please see the stack trace of the `check-types` command up above in the previous comment - make these non-nullable causes errors which I am not sure how to solve.

Reply:
this has been fixed with the latest commit (thanks @devinbinnie )